components/connectivity-adapter/internal/appregistry/service/manager.go

```diff
@@ -80,7 +80,39 @@ func (s *serviceManager) Create(serviceDetails model.GraphQLServiceDetailsInput)
 }
 
 func (s *serviceManager) GetFromApplicationDetails(serviceID string) (model.GraphQLServiceDetails, error) {
-	panic("implement me")
+	serviceRef, err := s.appLabeler.ReadServiceReference(s.appDetails, serviceID)
+	if err != nil {
+		return model.GraphQLServiceDetails{}, err
+	}
+	var outputApi *graphql.APIDefinitionExt
```

Comment:
I would move the logic to Labeler. Actually, there is a method for that - `ReadService(appDetails graphql.ApplicationExt, serviceID string) (GraphQLServiceDetails, error)`